yeat/Snakefile

```diff
@@ -28,8 +28,18 @@ rule copyinput:
         read1=config["read1"],
         read2=config["read2"]
     run:
-        copyfile(input.read1, output.read1)
-        copyfile(input.read2, output.read2)
+        if input.read1[-3:] == ".gz" and input.read2[-3:] == ".gz":
+            copyfile(input.read1, output.read1)
+            copyfile(input.read2, output.read2)
+        elif input.read1[-3:] != ".gz" and input.read2[-3:] == ".gz":
+            shell("gzip -c {input.read1} > {output.read1}")
+            copyfile(input.read2, output.read2)
```

Comment:
A couple of comments.

1. You might be interested in the `str.endswith()` method, e.g. `filename.endswith(".gz")`
2. Is there any particular benefit or reason you're handling each possible combination of gzipped and non-gzipped input files, rather than handling each file on its own? If it works, something like the following would be a more clear and concise way to handle this.

```python
for inreads, outreads in zip(input, output):
    if inreads.endswith(".gz"):
        copyfile(inreads, outreads)
    else:
        shell(f"gzip -c {inreads} > {outreads}")
```

Reply:
These are great suggestions! I didn't realize that we could `zip()` our `input` and `outputs` like that.